stripe-core/src/main/java/com/stripe/android/core/injection/Injectable.kt

```diff
@@ -6,14 +6,19 @@ import androidx.annotation.RestrictTo
  * Mark a class that can be injected by a [Injector].
  * This should be implemented by classes that cannot directly have their dependencies injected
  * through constructor and need to have them injected through lateinit properties.
+ *
+ * @param <FallbackInitializeParam> Class that holds the parameters required to recreate the Dagger
+ * dependency graph during fallback.
+ * @param <InjectorType> Injector class created during fallback, and used to inject into other
+ * classes. [Unit] if the injector created during fallback does not inject into other classes.
  */
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-interface Injectable<FallbackInitializeParam> {
+interface Injectable<FallbackInitializeParam, InjectorType> {
     /**
      * Fallback initialization logic for the dependencies when [Injector] is not available. This
      * could happen when the app process is killed and static [Injector]s are cleared up.
      *
      * An [Injectable] should check when this happens and calls this function manually.
      */
-    fun fallbackInitialize(arg: FallbackInitializeParam)
+    fun fallbackInitialize(arg: FallbackInitializeParam): InjectorType
```

Comment:
Could we expose a separate `Injectable.getInjector` method? having it returned here seems a bit hacky as some method would not return anything

Reply:
The problem is that we only know the injector at the time the `Injectable` itself is being injected (either from `WeakMapInjectorRegistry` or creating a new one in `fallbackInitialize`). We could just set a local variable [like it's currently done](https://github.com/stripe/stripe-android/blob/e84d6295fc27cb2ab589dbc0119df735ef5536f1/paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt#L335-L363), but I wanted to make that part of the injection itself. We can chat tomorrow about it, I arrived to this as the best option.